File: A4KPI/Models/Objective.cs
using A4KPI.Models.Interface;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace A4KPI.Models
{

    [Table("Objectives")]
    public class Objective : IDateTracking
    {
        [Key]
        public int Id { get; set; }
        [MaxLength(1000)]
        public string Topic { get; set; }
        public bool Status { get; set; }
        public int CreatedBy { get; set; }
        public int? ModifiedBy { get; set; }
        public DateTime Date { get; set; }
        public DateTime CreatedTime { get; set; }
        public DateTime? ModifiedTime { get; set; }
        public virtual ICollection<ToDoList> ToDoList { get; set; }
        public virtual ICollection<ResultOfMonth> ResultOfMonth { get; set; }
        public virtual ICollection<PIC> PICs { get; set; }
        [ForeignKey(nameof(CreatedBy))]
        public virtual Account Creator { get; set; }

        public virtual ICollection<Performance> Performances { get; set; }


    }
}

File: App/src/app/_core/_model/objective.ts
import { Account } from "./account";

export interface Objective {
  id: number;
  topic: string;
  status: boolean;
  accountId: number;
  createdBy: number;
  modifiedBy: number | null;
  createdTime: string;
  date: string;
  modifiedTime: string | null;
  account: Account;
  accounts: string;
  accountIdList: number[];

}

export interface ObjectiveRequest {
  id: number;
  topic: string;
  status: boolean;
  date: string;
  accountIdList: number[];
}

